Code/Framework/AzToolsFramework/AzToolsFramework/Manipulators/PaintBrushManipulator.cpp

```diff
@@ -159,84 +190,162 @@ namespace AzToolsFramework
         }
 
         // We found a collision point, so move the paintbrush.
-        m_center = worldSurfacePosition.value();
-        AZ::Transform space = AZ::Transform::CreateTranslation(m_center);
+        AZ::Vector3 brushCenter = worldSurfacePosition.value();
+        AZ::Transform space = AZ::Transform::CreateTranslation(brushCenter);
         SetSpace(space);
         PaintBrushNotificationBus::Event(m_ownerEntityComponentId, &PaintBrushNotificationBus::Events::OnWorldSpaceChanged, space);
 
         // If we're currently painting, send off a paint notification.
         if (m_isPainting)
         {
-            // Keep track of the previous painted point and the current one. We'll use that to create a brush stroke
-            // that covers everything in-between.
+            AZ::Vector2 currentCenter2D(brushCenter);
+
+            // Get our current paint brush settings.
+
+            PaintBrushSettings currentSettings;
+            PaintBrushSettingsRequestBus::BroadcastResult(currentSettings, &PaintBrushSettingsRequestBus::Events::GetSettings);
+
+            // Early out if we're completely transparent, there's no distance between brush stamps, or the brush stamp size is 0.
+            if ((currentSettings.GetOpacityPercent() == 0.0f) ||
+                (currentSettings.GetFlowPercent() == 0.0f) ||
+                (currentSettings.GetDistancePercent() == 0.0f) ||
+                (currentSettings.GetSize() == 0.0f))
+            {
+                return;
+            }
+
+            // Convert our settings into the 0-1 range
+            const float hardness = currentSettings.GetHardnessPercent() / 100.0f;
+            const float flow = currentSettings.GetFlowPercent() / 100.0f;
+
+            // Get the distance between each brush stamp in world space.
+            const float distanceBetweenBrushStamps = currentSettings.GetSize() * (currentSettings.GetDistancePercent() / 100.0f);
+
+            // Track the list of center points for each brush stamp to draw for this mouse movement.
+            AZStd::vector<AZ::Vector2> brushStamps;
+
+            // If this is the first point that we're painting, add this location to the list of brush stamps and use it
+            // as the starting point.
             if (isFirstPaintedPoint)
             {
-                m_previousCenter = m_center;
+                brushStamps.emplace_back(currentCenter2D);
+                m_lastBrushCenter = currentCenter2D;
+                m_distanceSinceLastDraw = 0.0f;
             }
 
-            // Get our current paint brush settings.
+            // Get the direction that we've moved the mouse since the last mouse movement we handled.
+            AZ::Vector2 direction = (currentCenter2D - m_lastBrushCenter).GetNormalized();
```

Comment:
question: There may be situations potentially where this are the same and normalizing the direction fails? Might we want to use the safe version or early out if `currentCenter2d-m_lastBrusCenter` is zero?

Reply:
There _shouldn't_ be a situation that can cause that, but I admit I was being too subtle and clever for my own good here. Basically, if the distance is 0, then the totalDistance will be < distanceBetweenBrushStamps, and the direction will never get used. So it _ought_ to be safe, but there aren't any guard rails here.

I'll change it to use GetNormalizedSafe() just to be 100% safe.

Reply:
Cool thanks!